scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RemoveScala3OptionalBraces.scala

```diff
@@ -134,7 +154,73 @@ private class RemoveScala3OptionalBraces(implicit val ftoks: FormatTokens)
         else if (ftoks.prevNonComment(ft).left.is[Token.Equals]) removeToken
         else null
       case p: Tree.WithBody => if (p.body eq tree) removeToken else null
+      case p: Term.ArgClause =>
+        p.tokens.head match {
+          case px: Token.LeftBrace =>
+            onLeftForArgClause(p, Some(px eq ft.right))
+          case px: Token.LeftParen
+              if shouldRewriteArgClauseWithLeftParen[RedundantParens](px) =>
+            onLeftForArgClause(p, Some(true))
+          case _ => null
+        }
       case _ => null
     }
 
+  private def shouldRewriteArgClauseWithLeftParen[A <: Rule](
+      lp: Token
+  )(implicit ft: FormatToken, session: Session, tag: ClassTag[A]) = {
+    val prevFt = ftoks.prevNonComment(ft)
+    prevFt.left.eq(lp) && session
+      .claimedRule(prevFt.meta.idx - 1)
+      .exists(x => tag.runtimeClass.isInstance(x.rule))
+  }
+
+  def onLeftForArgClause(
+      tree: Term.ArgClause,
+      sharesBracesWithArg: Option[Boolean]
+  )(implicit ft: FormatToken, style: ScalafmtConfig): Replacement = {
+    val rob = style.rewrite.scala3.removeOptionalBraces
+    val maxStats = rob.fewerBracesMaxStats
+    if (maxStats == 0) return null
+    @tailrec
+    def checkCountInRange(queue: mutable.Queue[Tree], cnt: Int): Boolean =
+      if (cnt > maxStats) false
+      else if (queue.isEmpty) cnt >= rob.fewerBracesMinStats
+      else {
+        val next = queue.dequeue()
+        def enqueue(trees: Iterable[Tree]*): Int = {
+          val len = queue.length
+          trees.foreach(queue ++= _)
+          queue.length - len
+        }
+        val delta = next match {
+          case x: Term.Block => enqueue(x.stats)
+          case x: Tree.WithBody => enqueue(x.body :: Nil) - 1
+          case x: Tree.WithCases => enqueue(x.cases)
```

Comment:
you mentioned not counting cases in a match statement. this line applies to partial functions as well, should we not count them either?

Reply:
i have decided to go with the same approach that is used for [`runner.optimizer.forceConfigStyleOnOffset` parameter](https://scalameta.org/scalafmt/docs/configuration.html#forcing-config-style), by counting total span of non-whitespace tokens.

that was the only approach that is easy to describe, easy to implement and is not subject to non-idempotence problems (since we know which tokens are being removed during the rewrite session).